pingrid.py

```diff
@@ -121,14 +121,14 @@ def interp_func(output_grids: Iterable[np.ndarray]) -> np.ndarray:
     return interp_func
 
 
-def create_interp(da: xr.DataArray, dims: Tuple[str, str]) -> FuncInterp2d:
-    x = da[dims[1]].values
-    y = da[dims[0]].values
+def create_interp(da: xr.DataArray) -> FuncInterp2d:
+    x = da["lon"].values
+    y = da["lat"].values
     input_grids = [
         (y[0], y[1] - y[0]),
         (x[0], x[1] - x[0]),
     ]  # require at least 2 points in each spatial dimension, and assuming that the grid is even
-    input_data = da.values
+    input_data = da.transpose("lat", "lon").values
```

Comment:
determine whether this line is indeed a no-op (it seems like the lat/lon dimensions should already be in the right order wrt one another and thus this line should do nothing). Also consider reordering dimensions in original file to make XY slices continugous/put T dimension before XY.